### ISSUE TITLE
error: release not found

### ISSUE THREAD

**Issue by alesmenzel** [getsentry/sentry-javascript-bundler-plugins#495]
### Environment

What version are you running? Etc.

webpack@5.89.0, @sentry/webpack-plugin@2.14.2 uploading to Sentry cloud

### Steps to Reproduce

```js
// Sentry plugin configuration
sentryWebpackPlugin({
	debug: true,
	sourcemaps: { assets: './dist/js/**' },
	release: {
		name: '<branch>_#<commit_hash>',
		deploy: {
			env: '<name>',
			url: '<url>',
		},
	},
})
```

### Expected Result

Successful upload of assets and source maps.

### Actual Result

```
> webpack --config ./build/webpack.production.config.js

[sentry-webpack-plugin] Debug: Could not determine debug ID from bundle. This can happen if you did not clean your output folder before installing the Sentry plugin. File will not be source mapped: <redacted>/dist/js/runtime-main.9be08bc5.js
// ... more of those - redacted
[sentry-webpack-plugin] Debug: Could not determine debug ID from bundle. This can happen if you did not clean your output folder before installing the Sentry plugin. File will not be source mapped: <redacted>/dist/js/runtime-main.ddae145d.js
> Found 56 files
> Analyzing 56 sources
> Analyzing completed in 0.091s
> Adding source map references
node:internal/errors:932
  const err = new Error(message);
              ^

Error: Command failed: <redacted>/node_modules/@sentry/cli-darwin/bin/sentry-cli releases finalize master_#<commit_hash|redacted>
error: release not found

Add --log-level=[info|debug] or export SENTRY_LOG_LEVEL=[info|debug] to see more output.
Please attach the full debug log to all bug reports.

    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:518:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5) {
  code: 1,
  killed: false,
  signal: null,
  cmd: '<redacted>/node_modules/@sentry/cli-darwin/bin/sentry-cli releases finalize master_#<commit_hash|redacted>'
}

Node.js v20.11.0
$ // the webpack build process exists here, but Sentry keeps on outputting stuff to console

> Bundling completed in 0.612s
> Bundled 56 files for upload
> Bundle ID: <bundle-id>
> Optimizing completed in 0.026s
> Uploading completed in 4.244s
> Uploaded files to Sentry
> Processing completed in 0.362s
> File upload complete (processing pending on server)
> Organization: <org|redacted>
> Project: <project|redacted>
> Release: master_#<commit_hash|redacted>
> Dist: None
> Upload type: artifact bundle

Source Map Upload Report
  Minified Scripts
    ~/0a80963b-f30e-43ec-8fe9-47046e23ff4b-66.js (sourcemap at 549.6572e883.chunk.js.map, debug id 0a80963b-f30e-43ec-8fe9-47046e23ff4b)
    // ... more of those
    ~/ff0bc065-3cfc-4715-90a4-56968bbd92b9-55.js (sourcemap at 459.b0d32cc7.chunk.js.map, debug id ff0bc065-3cfc-4715-90a4-56968bbd92b9)
Source Maps
    ~/0a80963b-f30e-43ec-8fe9-47046e23ff4b-66.js.map (debug id 0a80963b-f30e-43ec-8fe9-47046e23ff4b)
    // ... more of those
    ~/ff0bc065-3cfc-4715-90a4-56968bbd92b9-55.js.map (debug id ff0bc065-3cfc-4715-90a4-56968bbd92b9)
```

**Comment by alesmenzel** [comment]
It is a duplicate of #392 and #289, but both have been closed without a FIX.

**Comment by alesmenzel** [comment]
The `[sentry-webpack-plugin] Debug: Could not determine debug ID` went away after cleaning the dist/ before the build, so that's unrelated.

**Comment by rodolfoBee** [comment]
This error seems to be related to the release name containing `#`, i can easily reproduce and solve by changing it to `-` or another separator. 

Another workaround is to not finalise the release by setting [release.finalize](https://www.npmjs.com/package/@sentry/webpack-plugin#releasefinalize) to false

**Comment by lforst** [comment]
@alipek Can you also verify that this is due to some special characters in the release name?

**Comment by rodolfoBee** [comment]
> Another workaround is to not finalise the release by setting [release.finalize](https://www.npmjs.com/package/@sentry/webpack-plugin#releasefinalize) to false

Updating on the workaround: please try renaming the release. further testing with finalize:false had issues uploading source maps to my release.

**Comment by suekou** [comment]
I encountered the same error while using Next.js. In my case, the issue was that a team member had changed the project name in Sentry, and the `next.config.js` file created during the Sentry setup with `npx @sentry/wizard@latest -i nextjs` still referenced the old project name. If you've changed the project name, I recommend checking your configuration files to ensure they don't still contain the old project name.

**Comment by lforst** [comment]
@szokeasaurusrex maybe we can improve the error message in Sentry CLI?

Suggestion: 
> error: Release not found. Verify that you have configured the right organization and project for this operation.

**Comment by szokeasaurusrex** [comment]
@lforst good idea, I agree that many of the Sentry CLI error messages are too vague. https://github.com/getsentry/sentry-cli/issues/2111.
